Source/Transform/KTFractionalFFT.cc

```diff
@@ -0,0 +1,278 @@
+            KTINFO(evlog, "Calculated alpha = " << fAlpha);
+        }
+
+        double norm = 0.;                       // norm of the current TS value
+        double phase = 0.;                      // argument of current TS value
+        double q1 = tan( 0.5 * fAlpha ) * KTMath::Pi() / (double)(slice.GetSliceSize());
+        double q2 = sin( fAlpha ) * KTMath::Pi() / (double)(slice.GetSliceSize());
+        
+        for( unsigned iComponent = 0; iComponent < tsData.GetNComponents(); ++iComponent )
+        {
+            KTDEBUG(evlog, "Processing component: " << iComponent);
+
+            // get TS from data object and make the new TS
+            ts = dynamic_cast< KTTimeSeriesFFTW* >(tsData.GetTimeSeries( iComponent ));
+            if( ts == nullptr )
+            {
+                KTWARN(evlog, "Couldn't find time series object. Continuing to next component");
+                continue;
+            }
+            
+            // First chirp transform
+            for( unsigned iBin = 0; iBin < ts->GetNTimeBins(); ++iBin )
+            {
+                // Obtain norm and phase from components
+                norm = std::hypot( (*ts)(iBin)[0], (*ts)(iBin)[1] );
+                phase = atan2( (*ts)(iBin)[1], (*ts)(iBin)[0] );
+
+                // Shift phase
+                phase -= q1 * (double)iBin * (double)iBin;
+
+                // Assign components from norm and new phase
+                (*tsDup)(iBin)[0] = norm * cos( phase );
+                (*tsDup)(iBin)[1] = norm * sin( phase );
+            }
+
+            // Forward FFT
+            fs = fForwardFFT.Transform( tsDup );
+            fs->SetNTimeBins( slice.GetSliceSize() );
+
+            // Second chirp transform
+            for( unsigned iBin = 0; iBin < fs->GetNFrequencyBins(); ++iBin )
+            {
+                // Obtain norm and phase from components
+                norm = std::hypot( (*fs)(iBin)[0], (*fs)(iBin)[1] );
+                phase = atan2( (*fs)(iBin)[1], (*fs)(iBin)[0] );
+
+                // Shift phase
+                phase -= q2 * (double)iBin * (double)iBin;
+
+                // Assign components from norm and new phase
+                (*fs)(iBin)[0] = norm * cos( phase );
+                (*fs)(iBin)[1] = norm * sin( phase );
+            }
+
+            // Reverse FFT
+            newTS = fReverseFFT.TransformToComplex( fs );
+
+            // Third chirp transform
+            for( unsigned iBin = 0; iBin < newTS->GetNTimeBins(); ++iBin )
+            {
+                // Obtain norm and phase from components
+                norm = std::hypot( (*newTS)(iBin)[0], (*newTS)(iBin)[1] );
+                phase = atan2( (*newTS)(iBin)[1], (*newTS)(iBin)[0] );
+
+                // Shift phase
+                phase -= q1 * (double)iBin * (double)iBin;
+
+                // Assign components from norm and new phase
+                (*newTS)(iBin)[0] = norm * cos( phase );
```

Comment:
Is this really right, that we save the same things to `newTS` and `newFS`?

Reply:
Yes. This way the output can be used as either a TS or FS, each of which may be needed as it is really a mixed/rotated domain.